src/System.CommandLine.DragonFruit/CommandLine.cs

```diff
@@ -223,12 +223,13 @@ public static class CommandLine
                         }
                         else
                         {
-                            foreach (var argument in builder.Command.Arguments)
+                            for (var i = 0; i < builder.Command.Arguments.Count; i++)
                             {
+                                var argument = builder.Command.Arguments[i];
                                 if (string.Equals(
-                                        argument.Name,
-                                        kebabCasedParameterName, 
-                                        StringComparison.OrdinalIgnoreCase))
+                                    argument.Name,
+                                    kebabCasedParameterName,
+                                    StringComparison.OrdinalIgnoreCase))
```

Comment:
I'm just curious on how big of an improvement is this. Indexing the collection vs. boxing the enumerator.

Reply:
I don't love the code change, but cumulatively, it seems to be helping. This is one of the benchmarks I've been checking:

![image](https://user-images.githubusercontent.com/547415/104505548-e88ee180-5598-11eb-9fdb-03b9c635dade.png)

Reply:
The code being benchmarked is: 

```csharp
            var subcommandOne = new Command("subcommand-one")
            {
                new Option<int>(new[] { "-i", "--int" }),
                new Argument<string[]>("arg")
            };
            subcommandOne.Handler = CommandHandler.Create<int, string[]>((@int, arg) => { });

            var subcommandTwo = new Command("subcommand-two")
            {
                new Option<int>(new[] { "-i", "--int" }),
                new Argument<string[]>("arg")
            };
            subcommandTwo.Handler = CommandHandler.Create<int, string[]>((@int, arg) => { });
            
            var root = new RootCommand
            {
                subcommandOne,
                subcommandTwo
            };

            var parser = new CommandLineBuilder(root)
                         .UseDefaults()
                         .Build();
                
            await parser.InvokeAsync(new []{"subcommand-two","-i","123","one","two","three"});
```

Reply:
Is the benchmark code id the repo?

Reply:
Not currently. The existing benchmarks project has a large number of benchmarks but didn't seem suited to a more exploratory perf tuning approach with prior baselines. I've been tinkering with a notebook for this kind of thing, and am still really learning my way around BenchmarkDotNet.

Reply:
![image](https://user-images.githubusercontent.com/547415/104788462-0ac66200-5747-11eb-9fb8-ad04017c66ed.png)